File: batoman_api/controllers/igdb.py
from litestar import Controller, get
from ..util import guard_logged_in
from async_igdb import IGDBClient, PlatformModel
from ..models import CacheObject


class IGDBController(Controller):
    path = "/services/igdb"
    guards = [guard_logged_in]

    @get("/platforms")
    async def get_platform_list(self, igdb: IGDBClient) -> list[PlatformModel]:
        cached = CacheObject[list[PlatformModel]].from_cache("platforms-cache")
        if cached:
            return cached.data

        result = await igdb.resolve_links(await igdb.platforms.find(limit=500))
        CacheObject[list[PlatformModel]].to_cache("platforms-cache", result)
        return result

File: batoman_client/src/util/api/methods/igdb.ts
import { PlatformModel } from "@/models";
import { ApiConstructor } from "./base";

export function IGDBMixin<TBase extends ApiConstructor>(Base: TBase) {
    return class IGDBMixin extends Base {
        async get_available_platforms(): Promise<PlatformModel[]> {
            const result = await this.request<PlatformModel[]>(
                "/services/igdb/platforms",
            );
            if (result.success) {
                return result.data;
            }
            return [];
        }
    };
}
